{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-8facbbd9297e>\", line 2, in <module>\n",
      "    from stable_baselines import PPO2\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\__init__.py\", line 4, in <module>\n",
      "    from stable_baselines.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\__init__.py\", line 1, in <module>\n",
      "    from stable_baselines.a2c.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\a2c.py\", line 5, in <module>\n",
      "    import tensorflow as tf\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 99, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 941, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 955, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 665, in _load_unlocked\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 678, in exec_module\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-8facbbd9297e>\", line 2, in <module>\n",
      "    from stable_baselines import PPO2\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\__init__.py\", line 4, in <module>\n",
      "    from stable_baselines.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\__init__.py\", line 1, in <module>\n",
      "    from stable_baselines.a2c.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\a2c.py\", line 5, in <module>\n",
      "    import tensorflow as tf\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 99, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-8facbbd9297e>\", line 2, in <module>\n",
      "    from stable_baselines import PPO2\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\__init__.py\", line 4, in <module>\n",
      "    from stable_baselines.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\__init__.py\", line 1, in <module>\n",
      "    from stable_baselines.a2c.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\a2c.py\", line 5, in <module>\n",
      "    import tensorflow as tf\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 99, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 941, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 955, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 665, in _load_unlocked\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 678, in exec_module\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-8facbbd9297e>\", line 2, in <module>\n",
      "    from stable_baselines import PPO2\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\__init__.py\", line 4, in <module>\n",
      "    from stable_baselines.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\__init__.py\", line 1, in <module>\n",
      "    from stable_baselines.a2c.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\a2c.py\", line 5, in <module>\n",
      "    import tensorflow as tf\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 99, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-8facbbd9297e>\", line 2, in <module>\n",
      "    from stable_baselines import PPO2\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\__init__.py\", line 4, in <module>\n",
      "    from stable_baselines.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\__init__.py\", line 1, in <module>\n",
      "    from stable_baselines.a2c.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\a2c.py\", line 5, in <module>\n",
      "    import tensorflow as tf\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 99, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2895, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3072, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3282, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1211, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1169, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 316, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 350, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 1490, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 1448, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 941, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"<frozen importlib._bootstrap>\", line 994, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 971, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 955, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 665, in _load_unlocked\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 678, in exec_module\n",
      "  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3343, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-1-8facbbd9297e>\", line 2, in <module>\n",
      "    from stable_baselines import PPO2\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\__init__.py\", line 4, in <module>\n",
      "    from stable_baselines.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\__init__.py\", line 1, in <module>\n",
      "    from stable_baselines.a2c.a2c import A2C\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\stable_baselines\\a2c\\a2c.py\", line 5, in <module>\n",
      "    import tensorflow as tf\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 99, in <module>\n",
      "    from tensorflow_core import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\__init__.py\", line 28, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\importlib\\__init__.py\", line 126, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\__init__.py\", line 49, in <module>\n",
      "    from tensorflow.python import pywrap_tensorflow\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 74, in <module>\n",
      "    raise ImportError(msg)\n",
      "ImportError: Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ImportError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3263, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3360, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1193, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2895, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3072, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3282, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2047, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1436, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1336, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1211, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1150, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 451, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2044, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow.py\", line 58, in <module>\n",
      "    from tensorflow.python.pywrap_tensorflow_internal import *\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 28, in <module>\n",
      "    _pywrap_tensorflow_internal = swig_import_helper()\n",
      "  File \"C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\pywrap_tensorflow_internal.py\", line 24, in swig_import_helper\n",
      "    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 243, in load_module\n",
      "    return load_dynamic(name, filename, file)\n",
      "  File \"c:\\Users\\Aryaman\\anaconda3\\envs\\he2\\lib\\imp.py\", line 343, in load_dynamic\n",
      "    return _load(spec)\n",
      "ImportError: DLL load failed: The paging file is too small for this operation to complete.\n",
      "\n",
      "\n",
      "Failed to load the native TensorFlow runtime.\n",
      "\n",
      "See https://www.tensorflow.org/install/errors\n",
      "\n",
      "for some common reasons and solutions.  Include the entire stack trace\n",
      "above this error message when asking for help.\n"
     ]
    }
   ],
   "source": [
    "import gym \n",
    "from stable_baselines import PPO2\n",
    "from stable_baselines.common.vec_env import DummyVecEnv\n",
    "from stable_baselines.deepq.policies import LnMlpPolicy\n",
    "from stable_baselines.bench import Monitor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'tensorflow'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-1-4282574fb12a>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mnumpy\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mnp\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[1;32mimport\u001b[0m \u001b[0mtensorflow\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mtf\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'tensorflow'"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import tensorflow as tf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "env_name = \"LunarLander-v2\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\common\\tf_util.py:191: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\common\\tf_util.py:200: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\common\\policies.py:116: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\common\\input.py:25: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\common\\policies.py:561: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use keras.layers.flatten instead.\n",
      "WARNING:tensorflow:From C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\layers\\core.py:332: Layer.apply (from tensorflow.python.keras.engine.base_layer) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Please use `layer.__call__` method instead.\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\common\\tf_layers.py:123: The name tf.get_variable is deprecated. Please use tf.compat.v1.get_variable instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\common\\distributions.py:326: The name tf.random_uniform is deprecated. Please use tf.random.uniform instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\common\\distributions.py:327: The name tf.log is deprecated. Please use tf.math.log instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\ppo2\\ppo2.py:190: The name tf.summary.scalar is deprecated. Please use tf.compat.v1.summary.scalar instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\ppo2\\ppo2.py:198: The name tf.trainable_variables is deprecated. Please use tf.compat.v1.trainable_variables instead.\n",
      "\n",
      "WARNING:tensorflow:From C:\\Users\\Aryaman\\AppData\\Roaming\\Python\\Python36\\site-packages\\tensorflow_core\\python\\ops\\math_grad.py:1424: where (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use tf.where in 2.0, which has the same broadcast rule as np.where\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\ppo2\\ppo2.py:206: The name tf.train.AdamOptimizer is deprecated. Please use tf.compat.v1.train.AdamOptimizer instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\ppo2\\ppo2.py:240: The name tf.global_variables_initializer is deprecated. Please use tf.compat.v1.global_variables_initializer instead.\n",
      "\n",
      "WARNING:tensorflow:From c:\\Users\\Aryaman\\anaconda3\\envs\\he\\lib\\site-packages\\stable_baselines\\ppo2\\ppo2.py:242: The name tf.summary.merge_all is deprecated. Please use tf.compat.v1.summary.merge_all instead.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "env = gym.make(env_name)\n",
    "env = DummyVecEnv([lambda: env])\n",
    "model = PPO2(\"MlpPolicy\", env, verbose=1)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------------------------\n",
      "| approxkl           | 3.8300703e-05  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.00714       |\n",
      "| fps                | 172            |\n",
      "| n_updates          | 1              |\n",
      "| policy_entropy     | 1.3862562      |\n",
      "| policy_loss        | -0.00094320567 |\n",
      "| serial_timesteps   | 128            |\n",
      "| time_elapsed       | 0              |\n",
      "| total_timesteps    | 128            |\n",
      "| value_loss         | 241.45503      |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 1.6262538e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.0028        |\n",
      "| fps                | 656           |\n",
      "| n_updates          | 2             |\n",
      "| policy_entropy     | 1.3860456     |\n",
      "| policy_loss        | 0.00034085603 |\n",
      "| serial_timesteps   | 256           |\n",
      "| time_elapsed       | 0.745         |\n",
      "| total_timesteps    | 256           |\n",
      "| value_loss         | 248.2659      |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 6.3562984e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.014         |\n",
      "| fps                | 639            |\n",
      "| n_updates          | 3              |\n",
      "| policy_entropy     | 1.3860054      |\n",
      "| policy_loss        | -5.2697258e-05 |\n",
      "| serial_timesteps   | 384            |\n",
      "| time_elapsed       | 0.941          |\n",
      "| total_timesteps    | 384            |\n",
      "| value_loss         | 839.968        |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 4.34151e-06   |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.00901      |\n",
      "| fps                | 609           |\n",
      "| n_updates          | 4             |\n",
      "| policy_entropy     | 1.3860283     |\n",
      "| policy_loss        | -0.0004508237 |\n",
      "| serial_timesteps   | 512           |\n",
      "| time_elapsed       | 1.14          |\n",
      "| total_timesteps    | 512           |\n",
      "| value_loss         | 499.27646     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 1.3367891e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0044        |\n",
      "| fps                | 677            |\n",
      "| n_updates          | 5              |\n",
      "| policy_entropy     | 1.3858927      |\n",
      "| policy_loss        | -0.00022408727 |\n",
      "| serial_timesteps   | 640            |\n",
      "| time_elapsed       | 1.35           |\n",
      "| total_timesteps    | 640            |\n",
      "| value_loss         | 472.80118      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 5.8280293e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | 0.0032         |\n",
      "| fps                | 606            |\n",
      "| n_updates          | 6              |\n",
      "| policy_entropy     | 1.3857813      |\n",
      "| policy_loss        | -0.00029310246 |\n",
      "| serial_timesteps   | 768            |\n",
      "| time_elapsed       | 1.54           |\n",
      "| total_timesteps    | 768            |\n",
      "| value_loss         | 581.644        |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 4.8932193e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.00918      |\n",
      "| fps                | 428           |\n",
      "| n_updates          | 7             |\n",
      "| policy_entropy     | 1.3858029     |\n",
      "| policy_loss        | 2.7559232e-05 |\n",
      "| serial_timesteps   | 896           |\n",
      "| time_elapsed       | 1.76          |\n",
      "| total_timesteps    | 896           |\n",
      "| value_loss         | 520.62885     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.2676454e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0109       |\n",
      "| fps                | 670           |\n",
      "| n_updates          | 8             |\n",
      "| policy_entropy     | 1.3859243     |\n",
      "| policy_loss        | -0.0002920573 |\n",
      "| serial_timesteps   | 1024          |\n",
      "| time_elapsed       | 2.06          |\n",
      "| total_timesteps    | 1024          |\n",
      "| value_loss         | 1707.8221     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 8.194054e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.00208      |\n",
      "| fps                | 643           |\n",
      "| n_updates          | 9             |\n",
      "| policy_entropy     | 1.3854884     |\n",
      "| policy_loss        | -0.0022579716 |\n",
      "| serial_timesteps   | 1152          |\n",
      "| time_elapsed       | 2.25          |\n",
      "| total_timesteps    | 1152          |\n",
      "| value_loss         | 619.29047     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.6880634e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.0101        |\n",
      "| fps                | 589           |\n",
      "| n_updates          | 10            |\n",
      "| policy_entropy     | 1.3842437     |\n",
      "| policy_loss        | 0.000516196   |\n",
      "| serial_timesteps   | 1280          |\n",
      "| time_elapsed       | 2.45          |\n",
      "| total_timesteps    | 1280          |\n",
      "| value_loss         | 1675.5222     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 6.1481924e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | 0.00535        |\n",
      "| fps                | 711            |\n",
      "| n_updates          | 11             |\n",
      "| policy_entropy     | 1.3849963      |\n",
      "| policy_loss        | -0.00014147395 |\n",
      "| serial_timesteps   | 1408           |\n",
      "| time_elapsed       | 2.67           |\n",
      "| total_timesteps    | 1408           |\n",
      "| value_loss         | 378.6804       |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 9.352974e-06  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0101       |\n",
      "| fps                | 615           |\n",
      "| n_updates          | 12            |\n",
      "| policy_entropy     | 1.3857943     |\n",
      "| policy_loss        | -0.0002085339 |\n",
      "| serial_timesteps   | 1536          |\n",
      "| time_elapsed       | 2.85          |\n",
      "| total_timesteps    | 1536          |\n",
      "| value_loss         | 598.7611      |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 7.771661e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0177       |\n",
      "| fps                | 576           |\n",
      "| n_updates          | 13            |\n",
      "| policy_entropy     | 1.3855488     |\n",
      "| policy_loss        | -0.0013721221 |\n",
      "| serial_timesteps   | 1664          |\n",
      "| time_elapsed       | 3.06          |\n",
      "| total_timesteps    | 1664          |\n",
      "| value_loss         | 825.9362      |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 0.00014436946 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0195       |\n",
      "| fps                | 606           |\n",
      "| n_updates          | 14            |\n",
      "| policy_entropy     | 1.3846345     |\n",
      "| policy_loss        | -0.0025078745 |\n",
      "| serial_timesteps   | 1792          |\n",
      "| time_elapsed       | 3.28          |\n",
      "| total_timesteps    | 1792          |\n",
      "| value_loss         | 2188.1577     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 6.130403e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0115       |\n",
      "| fps                | 656           |\n",
      "| n_updates          | 15            |\n",
      "| policy_entropy     | 1.3847349     |\n",
      "| policy_loss        | -0.0010270481 |\n",
      "| serial_timesteps   | 1920          |\n",
      "| time_elapsed       | 3.49          |\n",
      "| total_timesteps    | 1920          |\n",
      "| value_loss         | 92.76782      |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 9.709302e-06  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0148       |\n",
      "| fps                | 673           |\n",
      "| n_updates          | 16            |\n",
      "| policy_entropy     | 1.3824047     |\n",
      "| policy_loss        | -0.0006849562 |\n",
      "| serial_timesteps   | 2048          |\n",
      "| time_elapsed       | 3.69          |\n",
      "| total_timesteps    | 2048          |\n",
      "| value_loss         | 416.96762     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 8.592401e-05   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0141        |\n",
      "| fps                | 666            |\n",
      "| n_updates          | 17             |\n",
      "| policy_entropy     | 1.3822125      |\n",
      "| policy_loss        | -0.00018849026 |\n",
      "| serial_timesteps   | 2176           |\n",
      "| time_elapsed       | 3.88           |\n",
      "| total_timesteps    | 2176           |\n",
      "| value_loss         | 408.78375      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 1.3295787e-05  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0202        |\n",
      "| fps                | 566            |\n",
      "| n_updates          | 18             |\n",
      "| policy_entropy     | 1.383573       |\n",
      "| policy_loss        | -0.00093077647 |\n",
      "| serial_timesteps   | 2304           |\n",
      "| time_elapsed       | 4.07           |\n",
      "| total_timesteps    | 2304           |\n",
      "| value_loss         | 446.1127       |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 3.0129882e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0179       |\n",
      "| fps                | 556           |\n",
      "| n_updates          | 19            |\n",
      "| policy_entropy     | 1.3820564     |\n",
      "| policy_loss        | 0.00030683633 |\n",
      "| serial_timesteps   | 2432          |\n",
      "| time_elapsed       | 4.3           |\n",
      "| total_timesteps    | 2432          |\n",
      "| value_loss         | 691.40094     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 0.00041082507 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.00107       |\n",
      "| fps                | 643           |\n",
      "| n_updates          | 20            |\n",
      "| policy_entropy     | 1.381656      |\n",
      "| policy_loss        | -0.0016941389 |\n",
      "| serial_timesteps   | 2560          |\n",
      "| time_elapsed       | 4.53          |\n",
      "| total_timesteps    | 2560          |\n",
      "| value_loss         | 1142.4346     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 6.464515e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.00577       |\n",
      "| fps                | 703           |\n",
      "| n_updates          | 21            |\n",
      "| policy_entropy     | 1.3838894     |\n",
      "| policy_loss        | -0.0011990076 |\n",
      "| serial_timesteps   | 2688          |\n",
      "| time_elapsed       | 4.73          |\n",
      "| total_timesteps    | 2688          |\n",
      "| value_loss         | 679.28766     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 1.4097575e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.016        |\n",
      "| fps                | 752           |\n",
      "| n_updates          | 22            |\n",
      "| policy_entropy     | 1.3813516     |\n",
      "| policy_loss        | -6.166322e-05 |\n",
      "| serial_timesteps   | 2816          |\n",
      "| time_elapsed       | 4.91          |\n",
      "| total_timesteps    | 2816          |\n",
      "| value_loss         | 312.19937     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 3.4909103e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0321        |\n",
      "| fps                | 656            |\n",
      "| n_updates          | 23             |\n",
      "| policy_entropy     | 1.3829395      |\n",
      "| policy_loss        | -4.3358654e-05 |\n",
      "| serial_timesteps   | 2944           |\n",
      "| time_elapsed       | 5.08           |\n",
      "| total_timesteps    | 2944           |\n",
      "| value_loss         | 348.2219       |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 1.955769e-05   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | 0.0138         |\n",
      "| fps                | 624            |\n",
      "| n_updates          | 24             |\n",
      "| policy_entropy     | 1.3795799      |\n",
      "| policy_loss        | -0.00012642943 |\n",
      "| serial_timesteps   | 3072           |\n",
      "| time_elapsed       | 5.28           |\n",
      "| total_timesteps    | 3072           |\n",
      "| value_loss         | 959.61456      |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 3.2258373e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0217       |\n",
      "| fps                | 633           |\n",
      "| n_updates          | 25            |\n",
      "| policy_entropy     | 1.381966      |\n",
      "| policy_loss        | 0.0011799107  |\n",
      "| serial_timesteps   | 3200          |\n",
      "| time_elapsed       | 5.49          |\n",
      "| total_timesteps    | 3200          |\n",
      "| value_loss         | 1484.7014     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 0.00012145153 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0165       |\n",
      "| fps                | 646           |\n",
      "| n_updates          | 26            |\n",
      "| policy_entropy     | 1.380271      |\n",
      "| policy_loss        | -0.0011144847 |\n",
      "| serial_timesteps   | 3328          |\n",
      "| time_elapsed       | 5.69          |\n",
      "| total_timesteps    | 3328          |\n",
      "| value_loss         | 257.70493     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.376519e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0142       |\n",
      "| fps                | 739           |\n",
      "| n_updates          | 27            |\n",
      "| policy_entropy     | 1.382602      |\n",
      "| policy_loss        | 3.5662204e-05 |\n",
      "| serial_timesteps   | 3456          |\n",
      "| time_elapsed       | 5.89          |\n",
      "| total_timesteps    | 3456          |\n",
      "| value_loss         | 218.98688     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.6830116e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0275       |\n",
      "| fps                | 739           |\n",
      "| n_updates          | 28            |\n",
      "| policy_entropy     | 1.3774618     |\n",
      "| policy_loss        | -0.0003660559 |\n",
      "| serial_timesteps   | 3584          |\n",
      "| time_elapsed       | 6.06          |\n",
      "| total_timesteps    | 3584          |\n",
      "| value_loss         | 379.36526     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 1.814933e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.00291       |\n",
      "| fps                | 656           |\n",
      "| n_updates          | 29            |\n",
      "| policy_entropy     | 1.3806742     |\n",
      "| policy_loss        | -0.0010713176 |\n",
      "| serial_timesteps   | 3712          |\n",
      "| time_elapsed       | 6.24          |\n",
      "| total_timesteps    | 3712          |\n",
      "| value_loss         | 47.950268     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 1.876834e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0201       |\n",
      "| fps                | 673           |\n",
      "| n_updates          | 30            |\n",
      "| policy_entropy     | 1.3829124     |\n",
      "| policy_loss        | 0.00044702715 |\n",
      "| serial_timesteps   | 3840          |\n",
      "| time_elapsed       | 6.43          |\n",
      "| total_timesteps    | 3840          |\n",
      "| value_loss         | 359.94205     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 7.4169297e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.000685      |\n",
      "| fps                | 680           |\n",
      "| n_updates          | 31            |\n",
      "| policy_entropy     | 1.3790084     |\n",
      "| policy_loss        | 0.00026742817 |\n",
      "| serial_timesteps   | 3968          |\n",
      "| time_elapsed       | 6.62          |\n",
      "| total_timesteps    | 3968          |\n",
      "| value_loss         | 1066.3499     |\n",
      "--------------------------------------\n",
      "-------------------------------------\n",
      "| approxkl           | 8.559486e-05 |\n",
      "| clipfrac           | 0.0          |\n",
      "| explained_variance | -0.0776      |\n",
      "| fps                | 673          |\n",
      "| n_updates          | 32           |\n",
      "| policy_entropy     | 1.3576257    |\n",
      "| policy_loss        | -0.000866011 |\n",
      "| serial_timesteps   | 4096         |\n",
      "| time_elapsed       | 6.81         |\n",
      "| total_timesteps    | 4096         |\n",
      "| value_loss         | 706.90344    |\n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      "| approxkl           | 9.441276e-05 |\n",
      "| clipfrac           | 0.0          |\n",
      "| explained_variance | -0.0172      |\n",
      "| fps                | 757          |\n",
      "| n_updates          | 33           |\n",
      "| policy_entropy     | 1.3736998    |\n",
      "| policy_loss        | 0.0006168142 |\n",
      "| serial_timesteps   | 4224         |\n",
      "| time_elapsed       | 7            |\n",
      "| total_timesteps    | 4224         |\n",
      "| value_loss         | 242.6715     |\n",
      "-------------------------------------\n",
      "-------------------------------------\n",
      "| approxkl           | 7.772847e-06 |\n",
      "| clipfrac           | 0.0          |\n",
      "| explained_variance | -0.0209      |\n",
      "| fps                | 757          |\n",
      "| n_updates          | 34           |\n",
      "| policy_entropy     | 1.3695776    |\n",
      "| policy_loss        | -0.000591167 |\n",
      "| serial_timesteps   | 4352         |\n",
      "| time_elapsed       | 7.17         |\n",
      "| total_timesteps    | 4352         |\n",
      "| value_loss         | 281.544      |\n",
      "-------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 8.461273e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0102       |\n",
      "| fps                | 684           |\n",
      "| n_updates          | 35            |\n",
      "| policy_entropy     | 1.376503      |\n",
      "| policy_loss        | -0.0016451678 |\n",
      "| serial_timesteps   | 4480          |\n",
      "| time_elapsed       | 7.34          |\n",
      "| total_timesteps    | 4480          |\n",
      "| value_loss         | 967.85706     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 0.00011667586 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0206       |\n",
      "| fps                | 646           |\n",
      "| n_updates          | 36            |\n",
      "| policy_entropy     | 1.3832412     |\n",
      "| policy_loss        | -0.001274672  |\n",
      "| serial_timesteps   | 4608          |\n",
      "| time_elapsed       | 7.53          |\n",
      "| total_timesteps    | 4608          |\n",
      "| value_loss         | 264.627       |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.0506068e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0204       |\n",
      "| fps                | 630           |\n",
      "| n_updates          | 37            |\n",
      "| policy_entropy     | 1.3802899     |\n",
      "| policy_loss        | -0.0014902987 |\n",
      "| serial_timesteps   | 4736          |\n",
      "| time_elapsed       | 7.73          |\n",
      "| total_timesteps    | 4736          |\n",
      "| value_loss         | 263.302       |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 1.2282945e-05  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0346        |\n",
      "| fps                | 731            |\n",
      "| n_updates          | 38             |\n",
      "| policy_entropy     | 1.371151       |\n",
      "| policy_loss        | -0.00066555466 |\n",
      "| serial_timesteps   | 4864           |\n",
      "| time_elapsed       | 7.93           |\n",
      "| total_timesteps    | 4864           |\n",
      "| value_loss         | 218.7103       |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 3.8334006e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.00259       |\n",
      "| fps                | 761            |\n",
      "| n_updates          | 39             |\n",
      "| policy_entropy     | 1.3752836      |\n",
      "| policy_loss        | -0.00021834904 |\n",
      "| serial_timesteps   | 4992           |\n",
      "| time_elapsed       | 8.11           |\n",
      "| total_timesteps    | 4992           |\n",
      "| value_loss         | 337.06506      |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 4.159168e-05  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.00651      |\n",
      "| fps                | 699           |\n",
      "| n_updates          | 40            |\n",
      "| policy_entropy     | 1.3706013     |\n",
      "| policy_loss        | 0.00022944831 |\n",
      "| serial_timesteps   | 5120          |\n",
      "| time_elapsed       | 8.28          |\n",
      "| total_timesteps    | 5120          |\n",
      "| value_loss         | 610.98474     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 5.6181593e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.00859      |\n",
      "| fps                | 663           |\n",
      "| n_updates          | 41            |\n",
      "| policy_entropy     | 1.3779508     |\n",
      "| policy_loss        | 0.00030690734 |\n",
      "| serial_timesteps   | 5248          |\n",
      "| time_elapsed       | 8.46          |\n",
      "| total_timesteps    | 5248          |\n",
      "| value_loss         | 201.98822     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 4.104006e-06   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0268        |\n",
      "| fps                | 673            |\n",
      "| n_updates          | 42             |\n",
      "| policy_entropy     | 1.3736653      |\n",
      "| policy_loss        | -0.00019903446 |\n",
      "| serial_timesteps   | 5376           |\n",
      "| time_elapsed       | 8.66           |\n",
      "| total_timesteps    | 5376           |\n",
      "| value_loss         | 743.98773      |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 3.8019843e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0061       |\n",
      "| fps                | 707           |\n",
      "| n_updates          | 43            |\n",
      "| policy_entropy     | 1.3823994     |\n",
      "| policy_loss        | -7.259427e-05 |\n",
      "| serial_timesteps   | 5504          |\n",
      "| time_elapsed       | 8.85          |\n",
      "| total_timesteps    | 5504          |\n",
      "| value_loss         | 302.91153     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 4.826943e-06  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.00146       |\n",
      "| fps                | 731           |\n",
      "| n_updates          | 44            |\n",
      "| policy_entropy     | 1.3600879     |\n",
      "| policy_loss        | 0.00011608214 |\n",
      "| serial_timesteps   | 5632          |\n",
      "| time_elapsed       | 9.03          |\n",
      "| total_timesteps    | 5632          |\n",
      "| value_loss         | 586.77356     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.2295859e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0118       |\n",
      "| fps                | 719           |\n",
      "| n_updates          | 45            |\n",
      "| policy_entropy     | 1.3819563     |\n",
      "| policy_loss        | -6.874575e-05 |\n",
      "| serial_timesteps   | 5760          |\n",
      "| time_elapsed       | 9.21          |\n",
      "| total_timesteps    | 5760          |\n",
      "| value_loss         | 323.5238      |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 7.8266885e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.00326       |\n",
      "| fps                | 711            |\n",
      "| n_updates          | 46             |\n",
      "| policy_entropy     | 1.3838298      |\n",
      "| policy_loss        | -0.00017773756 |\n",
      "| serial_timesteps   | 5888           |\n",
      "| time_elapsed       | 9.39           |\n",
      "| total_timesteps    | 5888           |\n",
      "| value_loss         | 149.34592      |\n",
      "---------------------------------------\n",
      "-------------------------------------\n",
      "| approxkl           | 7.67607e-07  |\n",
      "| clipfrac           | 0.0          |\n",
      "| explained_variance | -0.0127      |\n",
      "| fps                | 670          |\n",
      "| n_updates          | 47           |\n",
      "| policy_entropy     | 1.3830044    |\n",
      "| policy_loss        | -6.80543e-05 |\n",
      "| serial_timesteps   | 6016         |\n",
      "| time_elapsed       | 9.57         |\n",
      "| total_timesteps    | 6016         |\n",
      "| value_loss         | 256.15182    |\n",
      "-------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.9854803e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.021        |\n",
      "| fps                | 603           |\n",
      "| n_updates          | 48            |\n",
      "| policy_entropy     | 1.3765682     |\n",
      "| policy_loss        | -0.0002961167 |\n",
      "| serial_timesteps   | 6144          |\n",
      "| time_elapsed       | 9.76          |\n",
      "| total_timesteps    | 6144          |\n",
      "| value_loss         | 317.36047     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 1.5316176e-05  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0048        |\n",
      "| fps                | 673            |\n",
      "| n_updates          | 49             |\n",
      "| policy_entropy     | 1.3681328      |\n",
      "| policy_loss        | -0.00012344698 |\n",
      "| serial_timesteps   | 6272           |\n",
      "| time_elapsed       | 9.97           |\n",
      "| total_timesteps    | 6272           |\n",
      "| value_loss         | 154.38628      |\n",
      "---------------------------------------\n",
      "-------------------------------------\n",
      "| approxkl           | 3.745103e-05 |\n",
      "| clipfrac           | 0.0          |\n",
      "| explained_variance | 0.00321      |\n",
      "| fps                | 748          |\n",
      "| n_updates          | 50           |\n",
      "| policy_entropy     | 1.3805568    |\n",
      "| policy_loss        | 0.0009082279 |\n",
      "| serial_timesteps   | 6400         |\n",
      "| time_elapsed       | 10.2         |\n",
      "| total_timesteps    | 6400         |\n",
      "| value_loss         | 376.3354     |\n",
      "-------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 4.1526546e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.012        |\n",
      "| fps                | 630           |\n",
      "| n_updates          | 51            |\n",
      "| policy_entropy     | 1.3701977     |\n",
      "| policy_loss        | -6.440934e-05 |\n",
      "| serial_timesteps   | 6528          |\n",
      "| time_elapsed       | 10.3          |\n",
      "| total_timesteps    | 6528          |\n",
      "| value_loss         | 292.61884     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 3.604265e-06   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0581        |\n",
      "| fps                | 630            |\n",
      "| n_updates          | 52             |\n",
      "| policy_entropy     | 1.3744237      |\n",
      "| policy_loss        | -0.00020938006 |\n",
      "| serial_timesteps   | 6656           |\n",
      "| time_elapsed       | 10.5           |\n",
      "| total_timesteps    | 6656           |\n",
      "| value_loss         | 583.9646       |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 5.438805e-06   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0116        |\n",
      "| fps                | 603            |\n",
      "| n_updates          | 53             |\n",
      "| policy_entropy     | 1.3703045      |\n",
      "| policy_loss        | -0.00025797053 |\n",
      "| serial_timesteps   | 6784           |\n",
      "| time_elapsed       | 10.7           |\n",
      "| total_timesteps    | 6784           |\n",
      "| value_loss         | 451.66837      |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 3.0356488e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.000385      |\n",
      "| fps                | 633           |\n",
      "| n_updates          | 54            |\n",
      "| policy_entropy     | 1.3715944     |\n",
      "| policy_loss        | 0.0012720262  |\n",
      "| serial_timesteps   | 6912          |\n",
      "| time_elapsed       | 11            |\n",
      "| total_timesteps    | 6912          |\n",
      "| value_loss         | 561.6737      |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 2.3028879e-05  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.00276       |\n",
      "| fps                | 639            |\n",
      "| n_updates          | 55             |\n",
      "| policy_entropy     | 1.37418        |\n",
      "| policy_loss        | -0.00060808007 |\n",
      "| serial_timesteps   | 7040           |\n",
      "| time_elapsed       | 11.2           |\n",
      "| total_timesteps    | 7040           |\n",
      "| value_loss         | 217.23532      |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 4.3083586e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.00193       |\n",
      "| fps                | 707           |\n",
      "| n_updates          | 56            |\n",
      "| policy_entropy     | 1.3668076     |\n",
      "| policy_loss        | 0.00064528896 |\n",
      "| serial_timesteps   | 7168          |\n",
      "| time_elapsed       | 11.4          |\n",
      "| total_timesteps    | 7168          |\n",
      "| value_loss         | 735.4333      |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 2.183823e-05   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.00718       |\n",
      "| fps                | 656            |\n",
      "| n_updates          | 57             |\n",
      "| policy_entropy     | 1.3504891      |\n",
      "| policy_loss        | -9.1177644e-05 |\n",
      "| serial_timesteps   | 7296           |\n",
      "| time_elapsed       | 11.5           |\n",
      "| total_timesteps    | 7296           |\n",
      "| value_loss         | 302.66537      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 1.5167491e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | 0.000344       |\n",
      "| fps                | 511            |\n",
      "| n_updates          | 58             |\n",
      "| policy_entropy     | 1.3776971      |\n",
      "| policy_loss        | -0.00035591354 |\n",
      "| serial_timesteps   | 7424           |\n",
      "| time_elapsed       | 11.7           |\n",
      "| total_timesteps    | 7424           |\n",
      "| value_loss         | 267.9522       |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 1.8945134e-07 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -5.03e-05     |\n",
      "| fps                | 568           |\n",
      "| n_updates          | 59            |\n",
      "| policy_entropy     | 1.3802754     |\n",
      "| policy_loss        | -5.34181e-05  |\n",
      "| serial_timesteps   | 7552          |\n",
      "| time_elapsed       | 12            |\n",
      "| total_timesteps    | 7552          |\n",
      "| value_loss         | 253.41283     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 6.87863e-07   |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0259       |\n",
      "| fps                | 576           |\n",
      "| n_updates          | 60            |\n",
      "| policy_entropy     | 1.3793026     |\n",
      "| policy_loss        | -7.234409e-05 |\n",
      "| serial_timesteps   | 7680          |\n",
      "| time_elapsed       | 12.2          |\n",
      "| total_timesteps    | 7680          |\n",
      "| value_loss         | 809.47784     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 1.42738e-07    |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | 0.00299        |\n",
      "| fps                | 606            |\n",
      "| n_updates          | 61             |\n",
      "| policy_entropy     | 1.3802595      |\n",
      "| policy_loss        | -2.8967508e-05 |\n",
      "| serial_timesteps   | 7808           |\n",
      "| time_elapsed       | 12.4           |\n",
      "| total_timesteps    | 7808           |\n",
      "| value_loss         | 169.85255      |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 1.003194e-06  |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.00206       |\n",
      "| fps                | 589           |\n",
      "| n_updates          | 62            |\n",
      "| policy_entropy     | 1.3792965     |\n",
      "| policy_loss        | 4.0462124e-05 |\n",
      "| serial_timesteps   | 7936          |\n",
      "| time_elapsed       | 12.7          |\n",
      "| total_timesteps    | 7936          |\n",
      "| value_loss         | 353.3238      |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.6923646e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.0223        |\n",
      "| fps                | 475           |\n",
      "| n_updates          | 63            |\n",
      "| policy_entropy     | 1.3825889     |\n",
      "| policy_loss        | -0.0015459906 |\n",
      "| serial_timesteps   | 8064          |\n",
      "| time_elapsed       | 12.9          |\n",
      "| total_timesteps    | 8064          |\n",
      "| value_loss         | 230.96414     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 3.9219107e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0455       |\n",
      "| fps                | 579           |\n",
      "| n_updates          | 64            |\n",
      "| policy_entropy     | 1.3662908     |\n",
      "| policy_loss        | 0.00081302435 |\n",
      "| serial_timesteps   | 8192          |\n",
      "| time_elapsed       | 13.1          |\n",
      "| total_timesteps    | 8192          |\n",
      "| value_loss         | 504.86514     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 3.8018254e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.0593       |\n",
      "| fps                | 703           |\n",
      "| n_updates          | 65            |\n",
      "| policy_entropy     | 1.3795666     |\n",
      "| policy_loss        | 9.400421e-05  |\n",
      "| serial_timesteps   | 8320          |\n",
      "| time_elapsed       | 13.4          |\n",
      "| total_timesteps    | 8320          |\n",
      "| value_loss         | 299.68015     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 3.4401019e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | 0.0039         |\n",
      "| fps                | 744            |\n",
      "| n_updates          | 66             |\n",
      "| policy_entropy     | 1.3718996      |\n",
      "| policy_loss        | -0.00021724892 |\n",
      "| serial_timesteps   | 8448           |\n",
      "| time_elapsed       | 13.5           |\n",
      "| total_timesteps    | 8448           |\n",
      "| value_loss         | 278.91666      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 2.6140833e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0732        |\n",
      "| fps                | 707            |\n",
      "| n_updates          | 67             |\n",
      "| policy_entropy     | 1.3635224      |\n",
      "| policy_loss        | -0.00024839886 |\n",
      "| serial_timesteps   | 8576           |\n",
      "| time_elapsed       | 13.7           |\n",
      "| total_timesteps    | 8576           |\n",
      "| value_loss         | 266.14368      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 5.397144e-07   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | 0.00256        |\n",
      "| fps                | 636            |\n",
      "| n_updates          | 68             |\n",
      "| policy_entropy     | 1.3482411      |\n",
      "| policy_loss        | -8.2405866e-05 |\n",
      "| serial_timesteps   | 8704           |\n",
      "| time_elapsed       | 13.9           |\n",
      "| total_timesteps    | 8704           |\n",
      "| value_loss         | 528.3085       |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 2.0946504e-07  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | 0.0103         |\n",
      "| fps                | 526            |\n",
      "| n_updates          | 69             |\n",
      "| policy_entropy     | 1.3805696      |\n",
      "| policy_loss        | -0.00010898698 |\n",
      "| serial_timesteps   | 8832           |\n",
      "| time_elapsed       | 14.1           |\n",
      "| total_timesteps    | 8832           |\n",
      "| value_loss         | 211.03946      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 4.1342713e-07  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.21          |\n",
      "| fps                | 492            |\n",
      "| n_updates          | 70             |\n",
      "| policy_entropy     | 1.3752583      |\n",
      "| policy_loss        | -0.00014143344 |\n",
      "| serial_timesteps   | 8960           |\n",
      "| time_elapsed       | 14.3           |\n",
      "| total_timesteps    | 8960           |\n",
      "| value_loss         | 1103.0951      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 8.056373e-06   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.00412       |\n",
      "| fps                | 531            |\n",
      "| n_updates          | 71             |\n",
      "| policy_entropy     | 1.3557938      |\n",
      "| policy_loss        | -1.8014049e-05 |\n",
      "| serial_timesteps   | 9088           |\n",
      "| time_elapsed       | 14.6           |\n",
      "| total_timesteps    | 9088           |\n",
      "| value_loss         | 429.73373      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 3.119234e-06   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0279        |\n",
      "| fps                | 524            |\n",
      "| n_updates          | 72             |\n",
      "| policy_entropy     | 1.3820857      |\n",
      "| policy_loss        | -0.00018436671 |\n",
      "| serial_timesteps   | 9216           |\n",
      "| time_elapsed       | 14.9           |\n",
      "| total_timesteps    | 9216           |\n",
      "| value_loss         | 192.00974      |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 3.9077963e-06  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.157         |\n",
      "| fps                | 542            |\n",
      "| n_updates          | 73             |\n",
      "| policy_entropy     | 1.353988       |\n",
      "| policy_loss        | -0.00012423331 |\n",
      "| serial_timesteps   | 9344           |\n",
      "| time_elapsed       | 15.1           |\n",
      "| total_timesteps    | 9344           |\n",
      "| value_loss         | 1051.146       |\n",
      "---------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 8.557398e-06   |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.0958        |\n",
      "| fps                | 520            |\n",
      "| n_updates          | 74             |\n",
      "| policy_entropy     | 1.3762022      |\n",
      "| policy_loss        | -0.00067409384 |\n",
      "| serial_timesteps   | 9472           |\n",
      "| time_elapsed       | 15.3           |\n",
      "| total_timesteps    | 9472           |\n",
      "| value_loss         | 647.05743      |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.7604816e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | -0.00158      |\n",
      "| fps                | 524           |\n",
      "| n_updates          | 75            |\n",
      "| policy_entropy     | 1.3715776     |\n",
      "| policy_loss        | -0.0002252718 |\n",
      "| serial_timesteps   | 9600          |\n",
      "| time_elapsed       | 15.6          |\n",
      "| total_timesteps    | 9600          |\n",
      "| value_loss         | 151.11949     |\n",
      "--------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 2.2811626e-05 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.0137        |\n",
      "| fps                | 498           |\n",
      "| n_updates          | 76            |\n",
      "| policy_entropy     | 1.3677101     |\n",
      "| policy_loss        | -0.0009080395 |\n",
      "| serial_timesteps   | 9728          |\n",
      "| time_elapsed       | 15.8          |\n",
      "| total_timesteps    | 9728          |\n",
      "| value_loss         | 464.65814     |\n",
      "--------------------------------------\n",
      "---------------------------------------\n",
      "| approxkl           | 0.00045012275  |\n",
      "| clipfrac           | 0.0            |\n",
      "| explained_variance | -0.000405      |\n",
      "| fps                | 483            |\n",
      "| n_updates          | 77             |\n",
      "| policy_entropy     | 1.3633124      |\n",
      "| policy_loss        | -0.00085190346 |\n",
      "| serial_timesteps   | 9856           |\n",
      "| time_elapsed       | 16.1           |\n",
      "| total_timesteps    | 9856           |\n",
      "| value_loss         | 589.1221       |\n",
      "---------------------------------------\n",
      "--------------------------------------\n",
      "| approxkl           | 9.1178035e-06 |\n",
      "| clipfrac           | 0.0           |\n",
      "| explained_variance | 0.00298       |\n",
      "| fps                | 612           |\n",
      "| n_updates          | 78            |\n",
      "| policy_entropy     | 1.380053      |\n",
      "| policy_loss        | -0.000441676  |\n",
      "| serial_timesteps   | 9984          |\n",
      "| time_elapsed       | 16.4          |\n",
      "| total_timesteps    | 9984          |\n",
      "| value_loss         | 205.40817     |\n",
      "--------------------------------------\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<stable_baselines.ppo2.ppo2.PPO2 at 0x19c037bb978>"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "model.learn(10000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "model.save(\"PPO(mlp)-10000_model\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Episode:1 Score[-182.82509]\n",
      "Episode:2 Score[-99.25839]\n",
      "Episode:3 Score[-89.57]\n",
      "Episode:4 Score[-87.5442]\n",
      "Episode:5 Score[-65.63847]\n",
      "Episode:6 Score[-109.06319]\n",
      "Episode:7 Score[-120.189865]\n",
      "Episode:8 Score[-116.3168]\n",
      "avg score [-108.80076]\n"
     ]
    }
   ],
   "source": [
    "model = PPO2.load(\"PPO(mlp)-10000_model\", env = env)\n",
    "obs = env.reset()\n",
    "\n",
    "episodes = 10\n",
    "total_score =0\n",
    "for episode in range(1,episodes-1):\n",
    "    state = env.reset()\n",
    "    done = False\n",
    "    score = 0\n",
    "    \n",
    "    while not done:\n",
    "        env.render()\n",
    "        action, _states = model.predict(obs) \n",
    "        obs, reward, done, info = env.step(action)\n",
    "        score+= reward\n",
    "        total_score+=reward\n",
    "\n",
    "    print('Episode:{} Score{}'.format(episode,score))\n",
    "    \n",
    "print(\"avg score\", str(total_score/episode))\n",
    "env.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "ename": "",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31mThe Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click <a href='https://aka.ms/vscodeJupyterKernelCrash'>here</a> for more info. View Jupyter <a href='command:jupyter.viewOutput'>log</a> for further details."
     ]
    }
   ],
   "source": [
    "env.close()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "he2",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.13 |Anaconda, Inc.| (default, Mar 16 2021, 11:37:27) [MSC v.1916 64 bit (AMD64)]"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "4da64681f33cbef1aee73846e0608651d3110b5465e1ce25a3b4efceb34c727f"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
